### PR TITLE
Update react-resize-detector types to version 4

### PR DESCRIPTION
**Pull request by humphreybc**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/maslianok/react-resize-detector/releases/tag/v4.0.0
- [x] Increase the version number in the header if appropriate.

### CODE REVIEW

**Review by reminjp — APPROVED** [review]
LGTM. Thank you for updating.

File: types/react-resize-detector/index.d.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Type definitions for react-resize-detector 3.1
+// Type definitions for react-resize-detector 4.0
 // Project: https://github.com/maslianok/react-resize-detector
 // Definitions by: Matthew James <https://github.com/matthew-matvei>
```
Comment (rdrgn) [inline types/react-resize-detector/index.d.ts]: You can add your name if you want.

**Comment by rssfrncs** [comment]
Is this still correct as of version 4.0.5 of the library? 

I have 4.0.5 installed and it seems that `props.onResize` has the type `(width: number, height: number) => React.ReactNode` and not `(props: ReactResizeDetectorDimensions) => React.ReactNode)`.

**Comment by rdrgn** [comment]
Oops... I made a mistake.
`onResize` seems to be not changed. We have to revert the type definition.
https://github.com/maslianok/react-resize-detector/blob/396db3c6bb677042f9508c49942fe5c8a7f1b960/src/components/ResizeDetector.js#L98

**Comment by rssfrncs** [comment]
Fair enough it happens 👍

**Comment by rdrgn** [comment]
The type of `props.children` looks a bit complex...
https://github.com/maslianok/react-resize-detector/blob/396db3c6bb677042f9508c49942fe5c8a7f1b960/src/components/ResizeDetector.js#L163

**Comment by humphreybc** [comment]
Ohh… the releases docs definitely say it's changed to an object, but I just realized this is specifically the function-as-child pattern, not the `onResize` params. My bad!

// Before
<ReactResizeDetector handleWidth>
  {(width, height) => <div />}
</ReactResizeDetector>

// After
<ReactResizeDetector handleWidth>
  {({ width, height }) => <div />}
</ReactResizeDetector>

**Pull request merged**